Add ERC: Sovereign Bridged NFTs (#231)
ethereum/ERCs: ERCS/erc-7611.md

     */
    function mint(address _user, uint256 _tokenId, string _tokenURI) external;

    /**
     * @notice Mints batch of non-fungible tokens to a user
     * @dev Can only be called by a bridge
     * @param _user The address of the user who needs tokens minted
     * @param _tokenIdList The list of specific tokens to mint to a user
     * @param _tokenURIList The list of metadata for each individual token
     */
    function mintBatch(address _user, uint256[] calldata _tokenIdList, string[] calldata _tokenURIList) external;

    /**
     * @notice Burns a non-fungible token for a user
     * @dev Can only be called by a bridge
     * @param _user The address of the user who needs to burn the non-fungible token
     * @param _tokenId The non-fungible token to burn
     */
    function burn(address _user, uint256 _tokenId) external;

    /**
     * @notice Burns non-fungible tokens for a user
     * @dev Can only be called by a bridge
     * @param _user The address of the user who needs tokens burned
     * @param _tokenIdList The list of non-fungible tokens to burn
     */
    function burnBatch(address _user, uint256[] calldata _tokenIdList) external;
}
```

> [ERC-721](./EIP-721) does not provide developers with batch transfer functionality -- this has been a point of pain for developers leading to the introduction of new standards like [ERC-1155](./EIP-1155). This problem is exacerbated even further when programming across multiple rollups as it introduces asynchrony and the cost of compensating an interoperability network for verifying and relaying the transaction. It is for this reason that we decided to extend the contract with additional `mintBatch` and `burnBatch` function interfaces.

Reviewer: ```suggestion
[ERC-721](./eip-721.md) does not provide developers with batch transfer functionality -- this has been a point of pain for developers leading to the introduction of new standards like [ERC-1155](./eip-1155.md). This problem is exacerbated even further when programming across multiple rollups as it introduces asynchrony and the cost of compensating an interoperability network for verifying and relaying the transaction. It is for this reason that we decided to extend the contract with additional `mintBatch` and `burnBatch` function interfaces.
```

   */
  function localSupply() external view returns (uin256);
}
```

# Rationale

## Standardizing Sovereignty

This proposed standard was strongly influenced by the prior work done by the Connext team that is laid out in EIP-7281 (xERC20). All of the bridge authorization interfaces remain identical to EIP-7281 so that developer tooling that is built to empower asset issuers to manage bridges can be utilized both for multi domain ERC-20s as well as ERC-721s.

This EIP in combination with EIP-7281 are just the beginning of an open suite of standards for the Ethereum ecosystem as we embrace a truly rollup centric ecosystem. By utilizing standardized interfaces we can not only establish more rigorously tested developer tooling that can be used for all digital assets, we also open the opportunity for other standards to be introduced that further improve upon the functionality available to issuers of digital assets without requiring bespoke integrations.

As asset issuers are adopting this standard they should refer to the sections in EIP-7281 regarding rate limits and edge case considerations for limits when initializing any rate limit functionality.

## Data Structure Differences

> Having said that, the data structures utilized for ERC-721s fundamentally differ from those of ERC-20s and therefore we needed to modify the interfaces to a degree. The necessary modifications exists in the mint and burn functions.

Reviewer: ```suggestion
Having said that, the data structures utilized for ERC-721s fundamentally differ from those of [ERC-20](./eip-20.md) and therefore we needed to modify the interfaces to a degree. The necessary modifications exists in the mint and burn functions.
```

In addition — there are already large existing marketplaces that facilitate the exchange of NFTs among users and it is essential that xERC-721s comply with the interfaces that these exchanges use to facilitate seamless integration into the rest of the tooling that has been built around NFTs already. This similarly applies to NFTFi protocols.

# Security Considerations
Specific attention was paid to minimizing the code necessary to empower NFT communities with this functionality — we believe strongly that minimal standards are the best way to empower people to harness new opportunities in a manner that promotes composability while minimizing the risk of any security compromises.

Specifically, the most clear security consideration that teams should take into account as they upgrade collections to xERC-721s is the choice they make when selecting interoperability networks that facilitate the transfers of individual NFTs. We urge communities of high value xERC-721s to even consider utilizing additional security measures such as utilizing an m of n multi signature confirmation from multiple bridge providers.

> # Reference Implementation
> TODO: after receiving reviews & feedback

Reviewer: ```suggestion
```

You can just leave this section out until you decide to implement it.

# Copyright

> Copyright and related rights waived via [CC0](https://eips.ethereum.org/LICENSE).

Reviewer: ```suggestion
Copyright and related rights waived via [CC0](../LICENSE.md).
```
